daemon/graphdriver/overlay2/overlay.go

```diff
@@ -343,6 +351,11 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 			if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
 				return err
 			}
+		} else if d.options.quota.Size > 0 {
+			// Set default container disk quota limit
+			if err := d.quotaCtl.SetQuota(dir, d.options.quota); err != nil {
+				return err
+			}
```

Comment:
One problem with the way this code is structured is that it doesn't allow the user to override the default size with 0 (i.e. no quota for container).

Another problem is that it is just unneeded code duplication.
How about setting 
driver.options.quota = d.options.quota
just before d.parseStorageOpt

This will also be generic enough to apply future defaults if Quota struct ever
gets values like soft limit, inode limit etc...